frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/ApplicationMaster.java

```diff
@@ -753,16 +672,36 @@ private void recoverTransitionTaskStateQueue() {
   private void addContainerRequest(TaskStatus taskStatus) throws Exception {
     String taskRoleName = taskStatus.getTaskRoleName();
     TaskStatusLocator taskLocator = new TaskStatusLocator(taskRoleName, taskStatus.getTaskIndex());
-    ContainerRequest request = setupContainerRequest(taskStatus);
+    Integer containerRequestTimeoutSec = CommonUtils.getRandomNumber(
+        conf.getLauncherConfig().getAmContainerRequestMinTimeoutSec(),
+        conf.getLauncherConfig().getAmContainerRequestMaxTimeoutSec());
 
```

Comment:
Only the request to the same node will be timeout.  Than, it is not a big big issue.

Reply:
For node specified request, yes, but for nodelabel specified request, no.

---
In reply to: [158221031](https://github.com/Microsoft/pai/pull/121#discussion_r158221031) [](ancestors = 158221031)